token-lending/program/src/processor.rs

```diff
@@ -1859,6 +1861,20 @@ fn spl_token_init_mint(params: TokenInitializeMintParams<'_, '_>) -> ProgramResu
     result.map_err(|_| LendingError::TokenInitializeMintFailed.into())
 }
 
+/// Invoke signed unless signers seeds are empty
+#[inline(always)]
+fn invoke_optionally_signed(
+    instruction: &Instruction,
+    account_infos: &[AccountInfo],
+    authority_signer_seeds: &[&[u8]],
+) -> ProgramResult {
+    if authority_signer_seeds.is_empty() {
+        invoke(instruction, account_infos)
+    } else {
+        invoke_signed(instruction, account_infos, &[authority_signer_seeds])
+    }
+}
+
```

Comment:
@jstarry this fixes the issue with the PDA.

Reply:
one minor thing for this is I was saying you can actually call invoke_signed still but you have to pass in &[] instead of &[authority_signer_seeds] so it might be easier to just change the way you were calling the functions which need to sign to pass in  &[&[&[u8]]] instead of  &[&[u8]] to match the signing function signature anyway. but maybe that has some other drawbacks im not seeing?

Reply:
Looks like the implementation of `invoke` just calls `invoke_signed` with an empty array anyway:
```rust
pub fn invoke(instruction: &Instruction, account_infos: &[AccountInfo]) -> ProgramResult {
    invoke_signed(instruction, account_infos, &[])
}
```